File: Matlab/buffer_circ.m
function [out, buffer] = buffer_circ(in,buffer,delay,n)
%% buffer_circular
len = length(buffer);                   % Length Buffer
indexC = mod(n-1,len) + 1;              % Circular Index
indexD = mod(n-delay-1,len) + 1;        % Delay Index
out = buffer(indexD,1);                   % Output
buffer(indexC,1) = x;                   % Buffer 
end

File: Python/buffer_circ.py
def buffer_circular(x,buffer,delay,n):
    length = len(buffer)
    indexC = np.mod(n-1,length) 
    indexD = np.mod(n-delay-1,length) 
    y[n] = buffer[indexD]
    buffer[indexC] = x[n]
    return y[n], buffer
